frontend/src/static/js/v_summary.js

```diff
@@ -428,13 +442,34 @@ window.vSummary = {
       return null;
     },
     updateSortSelection() {
-      if (this.filterGroupSelection === 'groupByAuthors' && this.filterSort === 'displayName') {
-        this.filterSort = 'searchPath';
-      } else if (this.filterGroupSelection === 'groupByRepos' && this.filterSort === 'searchPath') {
-        this.filterSort = 'displayName';
+      this.getOptionWithOrder();
+      // Update UI selection to change all illegal options
+      if (this.filterGroupSelection === 'groupByAuthors') {
+        if (!this.sortWithinGroupSelection || this.sortingWithinOption === 'name') {
+          this.sortWithinGroupSelection = 'searchPath asc';
+        }
+        if (this.sortingOption !== 'name') {
+          this.sortGroupSelection = 'name asc';
+        }
+      } else if (this.filterGroupSelection === 'groupByRepos') {
+        if (!this.sortWithinGroupSelection || this.sortingWithinOption === 'searchPath') {
+          this.sortWithinGroupSelection = 'name asc';
+        }
+        if (this.sortingOption !== 'searchPath') {
+          this.sortGroupSelection = 'searchPath asc';
+        }
+      } else if (this.filterGroupSelection === 'groupByNone') {
+        this.sortWithinGroupSelection = '';
       }
     },
+    getOptionWithOrder() {
+      [this.sortingOption, this.isSortingDsc] = this.sortGroupSelection.split(' ');
+      this.isSortingDsc = this.isSortingDsc === 'dsc';
```

Comment:
doesn't seems very nice to reassign the variable to become boolean.
Would something like this work?

`name` and `name dsc`

if just `name`, the `this.isSortingDsc` will automatically be null,
hence can just use `if (this.isSortingDsc)` to do the logic check.

Reply:
Yes, that will be much simpler.